File: backend/app/models/task.py
# Task Model
# ----------
# Fields: id, title, description, project_id (FK),
#         sprint_id (FK, nullable), assignee_id (FK),
#         reporter_id (FK), status (enum: BACKLOG/TODO/
#         IN_PROGRESS/IN_REVIEW/DONE), priority (enum:
#         CRITICAL/HIGH/MEDIUM/LOW), story_points,
#         due_date, tags (JSON), created_at, updated_at
#
# Relationships: project, sprint, assignee, reporter,
#                comments, attachments, activity_logs
#
# Placeholder — implementation pending.

File: frontend/src/types/task.ts
/* Task Types */

export type TaskStatus = "backlog" | "todo" | "in_progress" | "in_review" | "done";
export type TaskPriority = "critical" | "high" | "medium" | "low";
export type TaskType = "bug" | "feature" | "task" | "story" | "epic";

export interface Task {
  id: string;
  title: string;
  description: string | null;
  status: TaskStatus;
  priority: TaskPriority;
  type: TaskType;
  assignee_id: string | null;
  project_id: string;
  sprint_id: string | null;
  created_at: string;
  updated_at: string;
}

export interface TaskSummary {
  id: string;
  title: string;
  status: TaskStatus;
  priority: TaskPriority;
}
